test/CometFlexVotingFork.t.sol

```diff
@@ -0,0 +1,1036 @@
+      8, // decimals
+      700_000_000_000_000_000, // borrowCollateralFactor
+      770_000_000_000_000_000, // liquidateCollateralFactor
+      950_000_000_000_000_000, // liquidationFactor
+      1_200_000_000_000 // supplyCap
+    );
+    _assetConfigs[2] = AssetConfig(
+      weth, // asset
+      0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419, // priceFeed
+      18, // decimals
+      825_000_000_000_000_000, // borrowCollateralFactor
+      895_000_000_000_000_000, // liquidateCollateralFactor
+      950_000_000_000_000_000, // liquidationFactor
+      350_000_000_000_000_000_000_000 // supplyCap
+    );
+    _assetConfigs[3] = AssetConfig(
+      0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984, // asset, UNI
+      0x553303d460EE0afB37EdFf9bE42922D8FF63220e, // priceFeed
+      18, // decimals
+      750_000_000_000_000_000, // borrowCollateralFactor
+      810_000_000_000_000_000, // liquidateCollateralFactor
+      930_000_000_000_000_000, // liquidationFactor
+      2_300_000_000_000_000_000_000_000 // supplyCap
+    );
+    _assetConfigs[4] = AssetConfig(
+      0x514910771AF9Ca656af840dff83E8264EcF986CA, // asset, LINK
+      0x2c1d072e956AFFC0D435Cb7AC38EF18d24d9127c, // priceFeed
+      18, // decimals
+      790_000_000_000_000_000, // borrowCollateralFactor
+      850_000_000_000_000_000, // liquidateCollateralFactor
+      930_000_000_000_000_000, // liquidationFactor
+      1_250_000_000_000_000_000_000_000 // supplyCap
+    );
+    Configuration memory _config = Configuration(
+      COMPOUND_GOVERNOR,
+      0xbbf3f1421D886E9b2c5D716B5192aC998af2012c, // pauseGuardian
+      address(govToken), // baseToken
+      0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6, // baseTokenPriceFeed, using the chainlink
+        // USDC/USD price feed
+      0x285617313887d43256F852cAE0Ee4de4b68D45B0, // extensionDelegate
+      800_000_000_000_000_000, // supplyKink
+      1_030_568_239 * 60 * 60 * 24 * 365, // supplyPerYearInterestRateSlopeLow
+      12_683_916_793 * 60 * 60 * 24 * 365, // supplyPerYearInterestRateSlopeHigh
+      0, // supplyPerYearInterestRateBase
+      800_000_000_000_000_000, // borrowKink
+      1_109_842_719 * 60 * 60 * 24 * 365, // borrowPerYearInterestRateSlopeLow
+      7_927_447_995 * 60 * 60 * 24 * 365, // borrowPerYearInterestRateSlopeHigh
+      475_646_879 * 60 * 60 * 24 * 365, // borrowPerYearInterestRateBase
+      600_000_000_000_000_000, // storeFrontPriceFactor
+      1_000_000_000_000_000, // trackingIndexScale
+      0, // baseTrackingSupplySpeed
+      3_257_060_185_185, // baseTrackingBorrowSpeed
+      1_000_000_000_000, // baseMinForRewards
+      100_000_000, // baseBorrowMin
+      5_000_000_000_000, // targetReserves
+      _assetConfigs
+    );
+
+    cToken = new CometFlexVoting(_config, address(flexVotingGovernor));
+
+    cToken.initializeStorage();
+    // ========= END DEPLOY NEW COMET ========================
+
+    // TODO is there anything we need to do to make this an "official" Comet?
```

Comment:
Good idea 👍

Reply:
I looked into this a bit. [Here's the proposal](https://compound.finance/governance/proposals/152) where they began the migration from v2 to v3. You can see that at that point the v3 contracts had already been deployed, e.g. cUSDCv3 was deployed [~7mo](https://etherscan.io/tx/0xfe8e6819f1b43dedacf864d96beaf6b924b69796a28e8c4a5622f21b3cc83cc0) before this proposal landed (Aug '22 vs Mar '23).

The deployed contract is an OZ `TransparentUpgradeableProxy`, though, so the initial deployment tx doesn't tell us a ton about what (if anything) goes into making one of these things "official". And all [the proposal](https://compound.finance/governance/proposals/152) actually does is use the `Configurator` to set some configuration vars (supply caps, borrow speeds, price factors) and upgrade the proxy to a new contract. It's not pushing the cUSDCv3 contract to a whitelist or anything obvious that would encode it as being "official" or anything.

This is further confirmed if you look at the [`Configurator.deploy` function](https://github.com/compound-finance/comet/blob/d7b414d3dbe57f6e59c2c571b5a629ca7a6fec68/contracts/Configurator.sol#L329-L337), which is meant to deploy new Comet instances:

```solidity
    /**
     * @notice Deploy a new Comet implementation using the factory and Configuration for that Comet proxy
     * @dev Note: Callable by anyone
     */
    function deploy(address cometProxy) external returns (address) {
        address newComet = CometFactory(factory[cometProxy]).clone(configuratorParams[cometProxy]);
        emit CometDeployed(cometProxy, newComet);
        return newComet;
    }
```

This is `external` so anyone can call it. New Comet instances aren't whitelisted or anything. But you can see that the Configurator determines (a) what the proxy is and (b) what configuration new instances will have.

So basically it looks like the only thing that makes a given Comet proxy contract "official" is that it was deployed by the Configurator. Thus var there are only [14](https://dune.com/queries/2490944) "official" v3 Comet instances -- though to be honest I'm not sure why there are even that many since they all have to have identical configurations. To be able to deploy anything from the Configurator it has to be pre-approved by the [Governor](https://etherscan.io/address/0x6d903f6003cca6255D85CcA4D3B5E5146dC33925), i.e. Compound's timelock contract, in the form of adding entries to the Configurator's `factory` and `configuratorParams` mappings for the approved proxy address. As we can see from the Configurator's [transaction history](https://etherscan.io/address/0x316f9708bb98af7da9c68c1c3b5e79039cd336e3), this has only ever been done with one proxy address, cUSDCv3 (here is [the setFactory call](https://etherscan.io/tx/0xd28da29068f2141e947c233fb62f8787c54edf9c5dd253356002ebc64f8adfcf), here is [the setConfiguration call](https://etherscan.io/tx/0x74092c79d75e410f632a00155c23650bcecd800fad8ef1abd28448c3ade40b7b)).

So if we wanted to get really fancy we could do that: deploy a new proxy, prank Governor calls to `setFactory` and `setConfiguration` for it. But tbh I don't see what it gains for us.

Reply:
Great write up, thanks. Leaving unresolved for @apbendi to see since this is useful. I agree that probably isn't necessary